### DISCUSSION TITLE
How to implement the growth process using TimelineBuilder.addCuttingPlane?

### DISCUSSION THREAD

**Discussion by shenshengyi** [iTwin/itwinjs-core#3129] [Q&A] [answered]
hi, I am using TimelineBuilder.addCuttingPlane.
I want to implement a process of growing cylindrical elements gradually. But using this function, there is always the effect of sudden sudden change.
I would like to know how to correctly implement this gradient process? Do I need to use other functions together? thanks.

**Comment by pmconne** [comment, marked as answer]
Show your code please.

**Reply by shenshengyi** [reply]
const script = makeCustomScript(
    (timeline: RenderSchedule.ElementTimelineBuilder) => {
      //   timeline.addVisibility(1542358800, 100, 1);
      //   timeline.addVisibility(1542906000, 100, 1);
      //   //   timeline.addTransform(1542358800, Transform.identity);
      //   timeline.addColor(1542358800, new RgbColor(0, 255, 0), 1);
      //   timeline.addColor(1542906000, new RgbColor(239, 241, 253), 1);
      const CutPlaneStart = {
        position: new Point3d(21.84000587463379, 0, 0),
        direction: new Point3d(0, 0, 1),
        // visible: false,
        hidden: true,
      };
      timeline.addCuttingPlane(1542358800, CutPlaneStart, 1);
      const CutPlaneEnd = {
        position: new Point3d(21.84000587463379, 0, 4),
        direction: new Point3d(0, 0, 1),
        visible: true,
        // hidden: false,
      };
      timeline.addCuttingPlane(1542906000, undefined, 1);
    },

    modelIds,
    elementIds
  );
Here is the main code snippet, thanks.

**Reply by pmconne** [reply]
If you want a gradual transition between two timeline entries you must specify `Interpolation.Linear`, otherwise you'll default to `Interpolation.Step` which jumps abruptly from one timeline entry to the next.

**Reply by shenshengyi** [reply]
![image](https://user-images.githubusercontent.com/26344907/151559799-7743ba67-615a-47d0-a29d-cbfa243e92f5.png)
Thanks a lot, it works fine.
But the visible and hidden parameters are not easy to understand, I tried several times.

**Reply by pmconne** [reply]
The visible and hidden parameters are indeed confusing.
You should not use them unless you want us to ignore your cutting plane, which you probably do not want.

**Reply by shenshengyi** [reply]
OK，Thanks.
